File: client/GhostClient/Core/Network/Packets/MatchEndPacket.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using GhostClient.Core;

namespace Ghost.Core.Network.Packets
{
    public class MatchEndPacket : Packet
    {
        public MatchEndPacket(Stream stream) : base(stream)
        {
        }

        public MatchEndPacket(Stream stream, bool autoClose) : base(stream, autoClose)
        {
        }

        public MatchEndPacket(byte[] data) : base(data)
        {
        }

        protected override void OnHandlePacket()
        {
            bool isWinner = Consume(1).BoolValue;
            long matchId = Consume(8).LongValue;

            GameHandler.Game.EndMatch();
        }
    }
}

File: client/core/src/com/boxtrotstudio/ghost/client/network/packets/MatchEndPacket.kt
package com.boxtrotstudio.ghost.client.network.packets

import com.boxtrotstudio.ghost.client.Ghost
import com.boxtrotstudio.ghost.client.core.game.TemporaryStats
import com.boxtrotstudio.ghost.client.handlers.scenes.StatsScene
import com.boxtrotstudio.ghost.client.network.Packet
import com.boxtrotstudio.ghost.client.network.PlayerClient

class MatchEndPacket : Packet<PlayerClient>() {

    override fun handle() {
        val isWinner = consume(1).asBoolean()
        val matchId = consume(8).asLong()
        val chunkSize = consume(4).asInt()
        val stats = consume(chunkSize).`as`(TemporaryStats::class.java)

        val statScreen = StatsScene(stats.get(TemporaryStats.SHOTS_FIRED).toInt(), stats.get(TemporaryStats.SHOTS_HIT).toInt(), stats.get(TemporaryStats.HAT_TRICK) == 1L, stats.get(TemporaryStats.ITEM_USAGE).toInt())
        statScreen.requestOrder(-5)
        Ghost.getInstance().addScene(statScreen)

        client.game.endMatch()
    }
}
